Source/Core/Common/JsonUtil.h

```diff
@@ -24,3 +27,18 @@ picojson::array ToJsonArray(const Range& data)
 
   return result;
 }
+
+template <typename Type>
+Type ReadNumericOrDefault(const picojson::object& obj, const std::string& key,
+                          Type default_value = Type{})
+{
+  const auto it = obj.find(key);
+  if (it == obj.end())
+    return default_value;
+  if (!it->second.is<double>())
+    return default_value;
+  return MathUtil::SaturatingCast<Type, double>(it->second.get<double>());
```

Comment:
The template parameter order is written so that the source type can be deduced, `SaturatingCast<Type>(value)` is enough and makes it look like an actual cast.